src/components/activity/ActivityRegistrator.tsx

```diff
@@ -0,0 +1,188 @@
+import React from 'react';
+import { View, KeyboardAvoidingView, TouchableWithoutFeedback, Keyboard } from 'react-native';
+import { Text, IconButton, Avatar, useTheme } from 'react-native-paper';
+
+import Slider from '@react-native-community/slider';
+
+import { DatePicker } from '../DatePicker';
+import { TimePicker, getCurrentTimeRounded } from '../TimePicker';
+import { SuggestiveTextInput } from '../SuggestiveTextInput';
+
+import { useTranslation } from 'language/LanguageProvider';
+import Storage from 'storage';
+
+export const ActivityRegistrator = ({ route, navigation }: any) => {
+  // route.params contains information from activity screen
+
+  const lang = useTranslation();
+  const { iconSizes, colors } = useTheme();
+
+  const steps = 1;
+  const defaultChoise = {
+    value: lang.activityRegistratorActivityDefaultChoise,
+    isDefault: true
+  }
+  let choises = [ defaultChoise ];
+
+
+  const [values, modifyValues] = Storage.useValues();
+  const [activities, modifyActivities] = Storage.useActivities();
+  
+  const [fromTime, setFromTime] = React.useState(getCurrentTimeRounded(0, steps));
+  const [toTime, setToTime] = React.useState(getCurrentTimeRounded(1, steps));
+  
+  const [date, setDate] = React.useState(new Date());
+  
+  const [choise, setChoise] = React.useState(defaultChoise);
+  const [activityText, setActivityText] = React.useState('');
+
+  const [importance, setImportance] = React.useState(5); 
+  const [enjoyment, setEnjoyment] = React.useState(5);
+
+
+
+  // Find topics to choose from depending on input icon
+  const addTopicEntries = (topics: any) => {
+    for(let topicIndex = 0; topicIndex < topics.length; ++topicIndex) {
+      for(let entryIndex = 0; entryIndex < topics[topicIndex].entries.length; ++entryIndex){
```

Comment:
Is this a part of the standard we're currently using or is it just for consistency? It should be functionally equivalent and personally i prefer this style, but ofc consistency goes over personal prefference.

Reply:
Not part of any standard, it is just that I have never seen anyone write for-loops like that. The increment I can understand (but not because it is better in any way), but I think there should be a space after the for.

Reply:
Yes, the space should be there.
The reason for my preference of ++i is because how it translated into machine code. When using i++ we save the value into a temporary variable so that we still can access it before we increment it, ++i does not do this. Compiler now are smart enough to optimize this, but that's where my preference comes from :)